infra/terraform/global/s3.tf

```diff
@@ -15,3 +15,27 @@ resource "aws_s3_bucket" "terraform_state" {
         enabled = true
     }
 }
+
+resource "aws_kms_key" "s3_uploads" {
+    description = "Key for upload object encryption"
+    deletion_window_in_days = 10
+}
+
+resource "aws_s3_bucket" "uploads" {
+    bucket = "${var.uploads_bucket_name}"
+    region = "${var.region}"
+    acl = "private"
+
+    versioning {
+        enabled = true
+    }
+
+    server_side_encryption_configuration {
+        rule {
+            apply_server_side_encryption_by_default {
+                kms_master_key_id = "${aws_kms_key.s3_uploads.arn}"
```

Comment:
Another KMS stupid question: How does this work? If S3 is going to use this KMS key transparently on the server, this means S3 can access this key, do we need to explicitly give S3 permission to this key or something (via IAM)?

Reply:
@xoen @RobinL FYI, I tried using customer-managed KMS encryption keys on this bucket as per AWS's best practice advice, but this resulted in permission errors. Documentation on required permissions for this are unclear or hard to find, so I've changed back to S3-managed keys for now.